### PR TITLE
Refactor CommonJS rewriting to be a normal pass

### PR DESCRIPTION
**Pull request by ChadKillingsworth**
Reworks CommonJS module rewriting as well as imported script rewriting into normal passes. Moves them into the default pass configuration.

Rewriting JSON files will be moved in a separate PR.

### CODE REVIEW

File: test/com/google/javascript/jscomp/Es6RewriteScriptsToModulesTest.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+ */
+
+package com.google.javascript.jscomp;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.common.collect.ImmutableList;
+import com.google.javascript.rhino.Node;
+
+/** Unit tests for {@link RewriteJsonToModule} */
```
Comment (MatrixFrog) [inline test/com/google/javascript/jscomp/Es6RewriteScriptsToModulesTest.java]: `/** Unit tests for {@link Es6RewriteScriptsToModules} */`

File: src/com/google/javascript/jscomp/Es6RewriteScriptsToModules.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  /**
+   * Creates a new Es6RewriteModules instance which can be used to rewrite ES6 modules to a
+   * concatenable form.
+   */
+  public Es6RewriteScriptsToModules(AbstractCompiler compiler) {
+    this.compiler = compiler;
+  }
+
+  /** Return whether or not the given script node represents an ES6 module file. */
+  public static boolean isEs6ModuleRoot(Node scriptNode) {
```
Comment (blickly) [inline src/com/google/javascript/jscomp/Es6RewriteScriptsToModules.java]: Looks like now we have 3 definitions of this method.  I'd probably get rid of all but one.

File: src/com/google/javascript/jscomp/AbstractCompiler.java [hunk 3]
```diff
@@ -615,6 +615,9 @@ abstract void updateGlobalVarReferences(Map<Var, ReferenceCollection>
    */
   abstract ModuleLoader getModuleLoader();
 
+  /** Lookup the type of a module from its name. */
+  abstract CompilerInput.ModuleType getModuleTypeByName(String moduleName);
```
[the remark this answers is not in the document]
Comment (ChadKillingsworth) [reply src/com/google/javascript/jscomp/AbstractCompiler.java]: This actually stems from other work which has yet to be PR'd. It makes much more sense to use this directly off the input though, so I've switched it.

**Comment by ChadKillingsworth** [comment]
@blickly Changes made. Let me know when you are ready to import and I'll squash the commits.

**Comment by blickly** [comment]
LGTM, let's squash and import.

Thanks Chad!

**Comment by ChadKillingsworth** [comment]
@blickly Sqaushed

**Pull request closed without merging**